File: python/week03/backend/order_service/app/schemas.py
# backend/order_service/app/schemas.py

"""
Pydantic schemas for the Order Service API.

These schemas define the data structures used for validating incoming requests
(e.g., creating or updating orders) and for serializing outgoing responses
(e.g., returning order details). They ensure data integrity and provide
clear API contracts for the Order microservice.
"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class OrderCreate(BaseModel):
    product_id: int = Field(..., gt=0, description="ID of the product ordered")
    quantity: int = Field(..., gt=0, description="Quantity of the product ordered")
    customer_name: str = Field(..., min_length=1, max_length=255, description="Customer's name")


class OrderUpdate(BaseModel):
    quantity: int = Field(None, gt=0, description="New quantity for the order")
    customer_name: str = Field(None, min_length=1, max_length=255, description="Customer's name")


class OrderResponse(OrderCreate):
    order_id: int
    created_at: datetime | None = None
    updated_at: datetime | None = None

    class Config:
        from_attributes = True

File: nodejs/week03/backend/order_service/app/schemas.js
// app/schemas.js

function validateOrderCreate(body) {
  if (!body.customer_name || !body.product_id || !body.quantity) {
    return 'Missing required fields'
  }
  if (typeof body.quantity !== 'number' || body.quantity <= 0) {
    return 'Quantity must be a positive number'
  }
  return null
}

module.exports = { validateOrderCreate }
